deployment/network-operator/templates/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml

```diff
@@ -13,7 +13,7 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 */}}
-{{- if .Values.deployCR }}
+  {{- if and (or .Values.ofedDriver.deploy .Values.nvPeerDriver.deploy .Values.devicePlugin.deploy (and .Values.secondaryNetwork.deploy (or .Values.secondaryNetwork.cniPlugins.deploy .Values.secondaryNetwork.multus.deploy .Values.secondaryNetwork.ipamPlugin.deploy))) .Values.deployCR }}
```

Comment:
this look very complicate if which will be hard to maintain. 
Maybe we should deploy the CRD even if everything is false. 
will that work?

Reply:
The spec is empty which cause an error

Reply:
IIRC, K8s will fail creating a CR with an empty spec even if the  object schema allows it.

we could define a "dummy" field but i dont see an added value except solving the problem above.

perhaps add this to [_helpers.tpl](https://github.com/Mellanox/network-operator/blob/master/deployment/network-operator/templates/_helpers.tpl) with comment  and use it here so the template is kept clean ?

we would still need to maintain though...